### PR TITLE
Section 2 updates

### LINKED ISSUE #211: Section 2
**Organization / Program:** Slandala <br>**Section:** 2 <br>**PDF Page:** None<br>**PDF Line(s):** 407<br>**Comment:** Should the requirement to update with the CAB requirements be in 1.5 or 9.12?<br><br>**Suggested Change:** None

### LINKED ISSUE #212: Section 2
**Organization / Program:** Slandala <br>**Section:** 2 <br>**PDF Page:** None<br>**PDF Line(s):** 413<br>**Comment:** Should the requirement to audit be in Section 8?  A requirement to post eviedence of an audit might belong here<br><br>**Suggested Change:** None

### LINKED ISSUE #213: Section 2.2
**Organization / Program:** Slandala <br>**Section:** 2.2 <br>**PDF Page:** None<br>**PDF Line(s):** 427-430<br>**Comment:** A "redacted" CPS that discloses "everything" required by RFC 3647 seem to be slightly at odds with each other.<br><br>**Suggested Change:** None

### LINKED ISSUE #248: Section 2.1
**Organization / Program:** Scott S. Perry CPA, PLLC <br>**Section:** 2.1 <br>**PDF Page:** 24<br>**PDF Line(s):** 418<br>**Comment:** The CP is silent in its requirements for repository availability<br><br>**Suggested Change:** The Issuer CA shall ensure that its root Certificate and the revocation data for issued Certificates are available through a repository 24 hours a day, 7 days a week with a minimum of 99% availability overall per year with a scheduled 
down‐time that does not exceed 0.5% annually.

### LINKED ISSUE #271: Section 2
**Organization / Program:** FPKIMA <br>**Section:** 2 <br>**PDF Page:** 23<br>**PDF Line(s):** 4.6<br>**Comment:** The CP amy also change with trust store program updates<br><br>**Suggested Change:** Add trust store program updates

### LINKED ISSUE #308: Section 2.2
**Organization / Program:** FPKIMA <br>**Section:** 2.2 <br>**PDF Page:** 24-25<br>**PDF Line(s):** 427-437<br>**Comment:** This section includes a cut/paste from the CAB Forum requirements, but the section 2.2 of that document seems to indicate what information must be disclosed in certain sections of the CP or CPS, but that doesn't mean it must be included as a statement here, or that it was intended to be included. It would make more sense to address the policy or practice on processing CAA records in section 4.2.2, rather than just re-iterate that this must be covered in those sections.  Additionally. Section 8.1 does not list any specific audit schemes that would answer the requirement as to what can be disclosed.<br><br>**Suggested Change:** Replace with the following text (replace URLs as necessary):

The FPKIPA maintains a website, <insert URL>, which hosts the redacted Certification Practice Statements, as well as individual CA practice and audit results per Section 8 of this CP. Certification Practice Statements for issuing CAs document the procedures for reviewing CAA records, as well as the policy or practice on processing CAA records for Fully Qualified Domain Names. All information disclosures conform to RFC 3647.

Web pages that allow for testing certificate validation up to the Federal NPE Root CA can be found at:
• https://valid.fpki.gov
• https://revoked.fpki.gov
• https://expired.fpki.gov

### LINKED ISSUE #335: Section 2
**Organization / Program:** DoD <br>**Section:** 2 <br>**PDF Page:** 24<br>**PDF Line(s):** 400-416<br>**Comment:** The header text in section 2 has nothing to do with publication and repository responsibilities.<br><br>**Suggested Change:** Move CPS requirements to section 1.5.4, move text about self-assessment to section 8. Include in section 2 requirements for what has to be published by whom

### LINKED ISSUE #465: Section 2 Editorial
Additional comments received on Section 2

| Organization / Program | PDF Line | Comment | Change |  
| --- | --- | --- | --- |  
| FPKIMA|400-416|The text of the section, with the focus on the CA/Browser Forum makes the section read as if the CA/Browser Forum requirements are more important than the U.S. Government requirements (including those mandated by law.)  It comes across as odd for a Government document.  While the CP can still be written to meet the requirements of the CA/Browser Forum Baseline Requirements, it should not be necessary to call out the CAB Forum specifically (and certainly not by version number.)  Why not then include other specific vendor trust store requirements as well. Additionally, the first paragraph states that the CP will be updated at least every 365 days, so there is no reason that it could not be updated more frequently.  This makes the paragraph about changes to CAB Forum Baseline requirements with a suspense of earlier than 365 days redundant. Lines 413-416 belong in Section 8.|Change text to:<br>The Federal PKI Policy Authority will review and update this Certificate Policy at least every 365 days to ensure compliance with applicable requirements. After review and approval, the CP document version number and a dated changelog entry shall be added even if no changes were deemed necessary. <br>The review and update shall include any changes needed to address:<br>• U.S. Federal Government requirements, mission needs, and changes to procedures to support the requirements/mission<br>• CA/Browser Forum Baseline Requirements updates<br>• Vendor Trust Store Requirement updates<br>The FPKI Policy Authority is responsible for monitoring source requirements, and initiating necessary changes to ensure continued compliance within the required timeframes, as well as ensuring associated CA Certification Practice Statements (CPS) are updated.<br>Each CA shall develop, implement, enforce, and update at least every 365 days a CPS that describes in detail how the CA implements the requirements of this CP.<br>Move lines 413-416 to Section 8. |
| FPKIMA|405|US not consistent with previous use. |Change to U.S. |
| FPKIMA|418-419|Lines 418 and 419 do not offer any real value as the CA is already required to abide by the CP.|Remove lines 418 and 419. |
| DoD|400-412|The text belongs in Section 9.12.1|Move the text to Section 9.12.1 |
| DoD|422-423|The requirement to document URLs in CPS is an overkill since the URLs are in the certificate unless a CPS stating that these URLs are found in the appropriate fields in the certificates is acceptable.|Delete the sentence unless CPS stating that these URLs are found in the appropriate fields in the certificates is acceptable. |  
| NSA|400|Needs to be requirements language|replace 'will' with 'SHALL' in first sentence.  Capitalize SHALL in second sentence. |  
| DoD|428-429|The term "audit" is overloaded.|Replace "audit" with "compliance audit" |
| NSA|428-429|I'm ok with this (posting practices and audits) as long as it doesn't disclose redacted CPS information.| |
| DoD|430| |Replace "with or" with "with" |
| DoD|431|Need to spell out CAA|First time the acronym is used |
| DoD|430-433|There is no reason to cite these two items in particular.  A CPS will address all requirements of the CP.|delete the two sentences |
| DoD|439-440|The section should have timeliness of publication|Replace the text with "CP and CPS shall be published within thirty days after being approved." |
| NSA|447|There should be a requirement about protecting the Repository from unauthorized modification.|Add, "Repository information SHALL be protected from unauthorized modification." |
| Slandala|401|The CPS should be reviewed annually in general, not just to address CAB requirements.| |
| Slandala|424|The policy should identify how fast updates should be posted.| |
| Slandala|440|The Federal PKI Policy Authority and CAs shall update and publish the Certificate Policy and Certification Practices Statements in accordance with Section "2.2". (circular reference)| |

### PR DESCRIPTION
**Pull request by lachellel**
23 comments reviewed. 
22 accepted, duplicate, partial acceptance or OBE (one comment and change made another irrelevant)

closes #308 closes #335 closes #248 closes #271 closes #211 closes #212 closes #213 closes #465 (partial acceptance)

### CODE REVIEW

**Comment by lachellel** [comment]
@konklone PR to the PR!

Whichever is easiest to review...

File: certificate-policy.md [hunk 1]
```diff
@@ -168,42 +169,44 @@ The key words "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMME
...
 
 ## 2.1 Repositories
-The CA shall make revocation information for Subordinate Certificates and Subscriber Certificates available in accordance with this Policy.
+Each CA shall disclose the following information through a publicly accessible Repository: 
+- CA Certificates
+- Certificate Revocation Lists (CRLs)
+- Online Certificate Status Protocol responses
+- CPS documents
+- Subscriber Agreements
+- Audit Reports
```
Comment (konklone) [inline certificate-policy.md]: I think it would help to be explicit that these should be unredacted.

File: certificate-policy.md [hunk 2]
```diff
@@ -168,42 +169,44 @@ The key words "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMME
...
 
-The CA shall post all CA certificates and CRLs issued by the CA in a repository that is publicly accessible through all Uniform Resource Identifier (URI) references asserted in valid certificates issued by that CA. The CA shall document the Uniform Resource Identifiers for CA certificates and CRLs in the CPS.
+Each CAs shall ensure that its root Certificate and the revocation data for issued Certificates are available through a Repository 24 hours a day, 7 days a week with a minimum of 99% availability overall per year with a scheduled down time that does not exceed 0.5% annually.
```
Comment (konklone) [inline certificate-policy.md]: Then what we're really saying is `99.5%` availability, not `99%`.

File: certificate-policy.md [hunk 3]
```diff
@@ -168,42 +169,44 @@ The key words "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMME
...
-
+The FPKIPA shall publicly post this CP on \<INSERT URL>, ensuring it is readily accessible on a 24x7 basis.
+
+Each CA shall disclose the following information through a publicly accessible Repository: 
+- CA Certificates
+- Certificate Revocation Lists (CRLs)
+- Online Certificate Status Protocol responses
+- CPS documents
+- Subscriber Agreements
+- Audit Reports
```
Comment (konklone) [inline certificate-policy.md]: Same comment about unredacted.

File: certificate-policy.md [hunk 4]
```diff
@@ -168,42 +169,44 @@ The key words "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMME
...
 
-All CAs approved to issue a CA certificate shall post to the Repository any issued CA certificate as soon as possible after issuance but no later than 15 days after issuance.  The FPKIPA or designee shall disclose the CA certificate and submit the CA certificate to trust stores and applicable databases, such as the Common CA Database, within thirty (30) days of issuance.
+Root CA(s) shall post to the Repository any issued CA Certificate as soon as possible after issuance but no later than fifteen (15) days after issuance.  The FPKIPA or designee shall disclose and submit the CA Certificate, CPS, and Audit Report(s) to trust store operators and applicable databases, such as the Common CA Database, as required by the trust store operator policies.
 
 Each CA shall publish CRLs in accordance with Section 4.9.7.
 
 ## 2.4 Access controls on repositories
-Each CA shall make its Repository publicly available in a read-only manner.
+Each CA shall make its Repository publicly available in a read-only manner.  Repository information shall be protected by unauthorized modification.
```
Comment (konklone) [inline certificate-policy.md]: Grammar typo here: should be `protected from unauthorized`, not `protected by unauthorized`.

But also, why do we need this?

**Review by konklone — CHANGES_REQUESTED** [review]
At least one technical comment (the 99% figure) and typo that I think would be worth changing before merge. 

I think it'd also be good to be explicit about non-redacted versions of CPSes and audit docs.

**Comment by konklone** [comment]
(Also noting that this will a new PR will be needed to merge `section1` into `master` after merging this, since I merged the PRs in the wrong order.)

File: certificate-policy.md (hunk 2 above)
Comment (lachellel) [reply certificate-policy.md]: This was for two different submitted comments.   It's in a handful of other CPs (non-govt) as well as other govt CPs.    

It's the clause "scheduled down time" that is being used to distinguish and is from a historical approach.  For example, a _scheduled_ patching cycle versus _unscheduled_ downtime due to natural disasters or other factors.   

I agree with you though; it's confusing in **today's** technology environment.  I'll update and we'll see what everyone thinks.

File: certificate-policy.md (hunk 3 above)
Comment (lachellel) [reply certificate-policy.md]: addressed

File: certificate-policy.md (hunk 1 above)
Comment (lachellel) [reply certificate-policy.md]: addressed

**Comment by techliaison** [comment]
Eric –
I’m not sure how to find the actual comment/current text.
But I hope we are keeping in the fact that only redacted CPS will be made publicly available.
Each CPS should have enough detail to be useable by the CA, but we do not want to put specifics about products, IPS, physical location etc in a publicly accessible document.


[LL: Modified email reply to remove the email thread]

File: certificate-policy.md (hunk 2 above)
Comment (konklone) [reply certificate-policy.md]: Ah, you're right, I didn't notice the distinction, and I can see how you would make it. Going to straight 99.5% works, or maybe just a rephrasing to make it more obvious to dunces like me who read too fast. :)

**Review by konklone — APPROVED** [review]

**Comment by konklone** [comment]
Thanks for addressing my comments, @lachellel. Looks good to me. It sounds like the redaction discussion will be taken up later this week, so no need to make it a blocker for merge here.

**Pull request merged**